app/src/main/java/org/sagebionetworks/research/mpower/EntryFragment.kt

```diff
@@ -30,55 +30,58 @@
-import org.sagebionetworks.bridge.android.manager.AuthenticationManager;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import javax.inject.Inject;
-
-public class EntryFragment extends BridgeAccessFragment {
-    private static final Logger LOGGER = LoggerFactory.getLogger(EntryFragment.class);
+class EntryFragment : BridgeAccessFragment() {
 
     @Inject
-    AuthenticationManager authenticationManager;
+    lateinit var authenticationManager: AuthenticationManager
 
-    @Override
-    public void onRequireAuthentication() {
-        LOGGER.debug("Showing MpPhoneAuthActivity");
+    override fun onRequireAuthentication() {
+        LOGGER.debug("Showing MpPhoneAuthActivity")
 
-        startActivity(new Intent(getContext(), MpPhoneAuthActivity.class));
+        startActivity(Intent(context, MpPhoneAuthActivity::class.java))
     }
 
-    @Override
-    public void onRequireConsent() {
-        LOGGER.debug("Showing WebConsentFragment");
+    override fun onRequireConsent() {
+        LOGGER.debug("Showing WebConsentFragment")
 
-        getChildFragmentManager().beginTransaction()
-                .replace(R.id.container, new WebConsentFragment())
-                .commit();
+        childFragmentManager.beginTransaction()
+                .replace(R.id.container, WebConsentFragment())
+                .commit()
     }
 
-    @Override
-    public void onAccessGranted() {
-        LOGGER.debug("Showing MainFragment");
+    override fun onAccessGranted() {
+        LOGGER.debug("Showing MainFragment")
 
-        // requires participants to have clinical_consent data group
-        // web consent fragment itself uses a feature flag to determine where participants are sent
-        if (getResources().getBoolean(R.bool.require_clinical_consent)
-                && !authenticationManager.getUserSessionInfo()
-                .getDataGroups().contains("clinical_consent")) {
-            LOGGER.info("clinical_consent data group required, even for consented users");
-            onRequireConsent();
-            return;
+
+        if (isMissingClinicalConsent()) {
+            LOGGER.info("clinical_consent data group required, even for consented users")
+            onRequireConsent()
+            return
         }
 
-        getChildFragmentManager().beginTransaction()
-                .replace(R.id.container, new MainFragment())
-                .commit();
+        childFragmentManager.beginTransaction()
+                .replace(R.id.container, MainFragment())
+                .commit()
+    }
+
+    /**
+     * Requires participants to have clinical_consent data group
+     * Web consent fragment itself uses a feature flag to determine where participants are sent
+     */
+    fun isMissingClinicalConsent(): Boolean {
+        return resources.getBoolean(R.bool.require_clinical_consent)
+                && !(authenticationManager.userSessionInfo?.dataGroups ?: listOf()).contains("clinical_consent")
+    }
+
+    companion object {

```

Comment:
nit but companion objects usually go at the top like Java's public static finals